### ISSUE TITLE
How to use this package with Remix?

### ISSUE THREAD

**Issue by piotrkulpinski** [t3-oss/t3-env#144]
Hi,
I've been using this package for some time now in NextJS and it works great. 
I'd love to be able to use it in my Remix apps as well but I can't get it to work without some complicated workarounds [like this](https://dev.to/remix-run-br/type-safe-environment-variables-on-both-client-and-server-with-remix-54l5).

Anyone successfully implemented this in Remix? Would love some advice.

Thanks

**Comment by chungweileong94** [comment]
I've been away from Remix for quite some time now, but you should still be able to set it up on Remix using the core package

I will try to create an example when I free to do so, but you can refer to doc for now https://env.t3.gg/docs/core 🙂

**Comment by piotrkulpinski** [comment]
I did follow the docs on this, but the thing with remix is that the components are loaded both from the server and the client, so every time I import the `env.ts` file in a component, I get the `Uncaught ReferenceError: **process is not defined**` error. 

Any working example would be very helpful. 
Thanks!

**Comment by chungweileong94** [comment]
> I did follow the docs on this, but the thing with remix is that the components are loaded both from the server and the client

So if the env variable is being use in both server and client, the env should be defined under `client`.

> I get the `Uncaught ReferenceError: process is not defined` error

Unfortunately, I wasn't able to get the same error in my test, mind if you can create a repro? 
Could it be something related to this [issue](https://github.com/remix-run/remix/issues/7201)? I also know that they recently release an early version of Vite support, which could also be a reason why it doesn't work as expected.

**Comment by piotrkulpinski** [comment]
I don't want the env variable to be available in both server and client. I just wanted to share the same file for both types just like I did with NextJS.

I guess that's not going to work like this in Remix, so I've created 2 separate files as described in the docs: `env.server.ts` using `process.env` as runtimeEnv and `env.client.ts` using `import.meta.env` from Vite. 

Looks like the issue is related to the new Vite integration in Remix as I'm currently using it (https://github.com/remix-run/remix/issues/7934).

I'll close this now as the issues is clearly not related to this package. 
Thank you for your help.
